java/client/src/main/java/glide/connectors/handlers/PubSubMessageQueue.java

```diff
@@ -0,0 +1,55 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.connectors.handlers;
+
+import glide.api.models.PubSubMessage;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/** A FIFO message queue for {@link PubSubMessage}. */
+public class PubSubMessageQueue {
+    // fields are protected to ease testing
+    /** The queue itself. */
+    protected final ConcurrentLinkedDeque<PubSubMessage> messageQueue = new ConcurrentLinkedDeque<>();
+
+    /**
+     * A promise for the first incoming message. Returned to a user, if message queried in async
+     * manner, but nothing received yet.
+     */
+    protected CompletableFuture<PubSubMessage> firstMessagePromise = new CompletableFuture<>();
```

Comment:
Why not private?

Reply:
I'm checking this in UTs, there is a comment about it on top of the class

Reply:
Package-private is usually used for this purpose (as well as using the VisibleForTesting annotation).

Reply:
Fixed, thanks!